### ISSUE TITLE
Ubuntu-latest image 20221204.2 Git Workflows failing after azPSVersion update

### ISSUE THREAD

**Issue by BartDecker** [actions/runner-images#6753]
### Description

We use "latest" for the azPSVersion which gave use 7.5.0 until last week using 'ubuntu-latest' as a runner. (20221127.1). We have a nightly build running in multiple azure subscriptions for which the pwsh scripts itself did not change over the course of last week. The switch from working to not working came with the change of image version and the resulting change in the azPSVersion

I see 'latest' now gives us 9.0.1 (image 20221204.2)

The az pwsh command fail at a point where they are used in a very basic way. No complex syntax which would be different for the 7.5.0 version of the cmdlet compared to the 9.0.1 one. (Get-AzResourceGroup -Name xxxx) which make me think that the problem lies in the runner and not so much in the command. 

Running the workflow with azPSVersion '7.5.0' instead of 'latest' solves the issue.

failing cmdlets: Get-AzRoleAssignment, Get-AzResourceGroup, Get-AzEventGridSubscription

Some example errors:

'Get-AzResourceGroup' command was found in the module 'Az.Resources',
     | but the module could not be loaded. For more information, run
     | 'Import-Module Az.Resources

The 'Get-AzEventGridSubscription' command was found in the module
     | 'Az.EventGrid', but the module could not be loaded. For more
     | information, run 'Import-Module Az.EventGrid

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [x] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20221204.2 -> https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20221204.2

### Is it regression?

20221127.1

### Expected behavior

Commandlets loaded and enabled for usage within workflow

### Actual behavior

Commandlets are failing because they are not loaded

### Repro steps

1. use workflow with ubuntu-latest & azPSVersion on latest (az pwsh cmdlets failures)
2. use the same workflow with ubuntu-latest & azPSVersion on 7.5.0 (works)

**Comment by igorboskovic3** [comment]
Hi @BartDecker we will take a look.

**Comment by BartDecker** [comment]
> Hi @BartDecker we will take a look.

Much appreciated!

**Comment by igorboskovic3** [comment]
Hi @BartDecker , we updated Az module few weeks ago to "9.0.1"and deployed last week,  so now latest is "9.0.1".
You can see [here](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#az-powershell-modules). Feel free to reach us again if you have any questions left.

**Comment by BartDecker** [comment]
> Hi @BartDecker , we updated Az module few weeks ago to "9.0.1"and deployed last week, so now latest is "9.0.1". You can see [here](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#az-powershell-modules). If I understood right, you resolved your issue if you use 7.5.0? Feel free to reach us again if you have any questions left.

Yeah I don't have the error with 7.5.0 but still think the 9.0.1 errors are image related and not cmdlet related. Something in the image seems not the be loaded correctly when the runner spins up. 

The cmdlet usage is so basic that the error can't be explained by a different az pwsh version. For example the et-AzResourceGroup with only one parameter -Name xxx is so simple and is the same in version 7.5.0 and 9.0.1

**Comment by Alexey-Ayupov** [comment]
Hello @BartDecker, I have tried today and could not reproduce your issue. If I understood correctly, you are using "Azure/powershell@v1" action, supporting which is out of the scope of our team. 

We will close this issue for now. Feel free to reach us if you have any questions.

P.S. The test code snippet
`jobs:
  build:
    strategy:
      matrix:
        os: [ubuntu-22.04, ubuntu-20.04, windows-2019, windows-2022]
    runs-on: ${{ matrix.os }}
    steps:
      - uses: azure/login@v1
        with:
          creds: '{AZURE_CREDS}'
          enable-AzPSSession: true`
    
      - name: Azure PowerShell Action
        uses: Azure/powershell@v1
        with:
          inlineScript: |
            get-azcontext
            Get-AzResourceGroup
          azPSVersion: "latest"

**Comment by BartDecker** [comment]
> Hello @BartDecker, I have tried today and could not reproduce your issue. If I understood correctly, you are using "Azure/powershell@v1" action, supporting which is out of the scope of our team.
> 
> We will close this issue for now. Feel free to reach us if you have any questions.
> 
> P.S. The test code snippet `jobs: build: strategy: matrix: os: [ubuntu-22.04, ubuntu-20.04, windows-2019, windows-2022] runs-on: ${{ matrix.os }} steps: - uses: azure/login@v1 with: creds: '{AZURE_CREDS}' enable-AzPSSession: true`
> 
> ```
>   - name: Azure PowerShell Action
>     uses: Azure/powershell@v1
>     with:
>       inlineScript: |
>         get-azcontext
>         Get-AzResourceGroup
>       azPSVersion: "latest"
> ```

Hi @Alexey-Ayupov Thanks for testing. How do I need to read the "Azure/powershell@v1" remark? Is it not supported? Or not supported by your team?

Is using the ubuntu-22.04 in the above test the same as using ubuntu-latest? Or is the first referring to image 20221127.1?

**Comment by Alexey-Ayupov** [comment]
Hi @BartDecker, answering your first question, the "Azure/powershell@v1" does not support by our team. I believe the support team sits here - https://github.com/Azure/powershell.
Ubuntu-latest is referring to ubuntu-20.04 however we are in the middle of migrating this label and shortly the latest will be ubuntu-22.04. 
We will try to run the test one more time using a script instead inline commands.

**Comment by BartDecker** [comment]
After testing the root cause of the issue was using "latest" for azPSVersion which was updated from 7.5.0 to 9.0.1 in the runner image. This creates issues when in the inline script or pwsh script that is called by the inline script in a workflow, az modules are loaded which are not part of the main az pwsh package. 

For example Az.ResourceGraph. These az modules are not updated in parallel with main AZ pwsh modules and are therefore not always compatible. 

Until these modules are also updated the only workaround is to not use 'latest' for the azPSVersion but fix it to a version that works in combination with the az modules loaded in a specific job.
